### PR TITLE
Change all interfaceId fields to use custom.fields.merchantReference instead

### LINKED ISSUE #62: Extension incompatible with /me/payments endpoint
When trying to use this extension alongside the /me/payments endpoint in order to create and process a payment, the extension throws an error due to being unable to specify an interfaceId from the My Payments version of the endpoint. Did some quick testing and removed the interfaceId validation from the extension and generated a random UUID instead and then the payment processes fine. What would need to be done in order to get this working properly with the my payment endpoint? It seems like you'd want to be able to specify the interfaceId from the client side in order to get the benefit of de-duplicating payments.

### PR DESCRIPTION
**Pull request by zackpollard**
This resolves #62 

This work should change the reference field from interfaceId to a custom field specified on the payment object. If you could give some feedback on whether this is the correct approach, that'd be great. If you want the name of the field to be different too, I can make that change.

All the unit tests past, one test fails but that seems to be failing on master too right now so it seems to be unrelated.

### CODE REVIEW

File: extension/resources/payment-custom-types.json [hunk 1]
```diff
@@ -7,6 +7,17 @@
     "payment"
   ],
   "fieldDefinitions": [
+    {
+      "type": {
+        "name": "String"
+      },
+      "name": "interfaceId",
```
Comment (butenkor) [inline extension/resources/payment-custom-types.json]: Please rename custom field `interfaceId` to `reference` or `merchantReference` as `interfaceId` has actually different meaning and purpose. It is a unique payment ID which is provided by the payment provider and not the shop itself. Adyen unfortunately does have any generalised payment specific unique IDs but rather transaction specific one only.

File: extension/test/fixtures/payment-no-method.json [hunk 2]
```diff
@@ -15,6 +14,7 @@
       "key": "ctp-adyen-integration-payment-custom-type"
     },
     "fields": {
+      "interfaceId": "pspReference",
```
Comment (butenkor) [inline extension/test/fixtures/payment-no-method.json]: I know it was like that before but `pspReference` implies that it is provided by PSP which is not the case here, right? Maybe better `paymentNoMethodReference` or just the same as in other fixtures `paymentReferenceId`?

File: notification/test/resources/payment-custom-types.json [hunk 3]
```diff
@@ -7,6 +7,17 @@
     "payment"
   ],
   "fieldDefinitions": [
+    {
+      "type": {
+        "name": "String"
+      },
+      "name": "interfaceId",
```
Comment (butenkor) [inline notification/test/resources/payment-custom-types.json]: Same here.

Comment (zackpollard) [reply notification/test/resources/payment-custom-types.json]: This has now been changed

File: extension/resources/payment-custom-types.json (hunk 1 above)
Comment (zackpollard) [reply extension/resources/payment-custom-types.json]: This has now been changed

File: extension/test/fixtures/payment-no-method.json (hunk 2 above)
Comment (zackpollard) [reply extension/test/fixtures/payment-no-method.json]: Have changed this to paymentReferenceId like in the rest of the tests

File: extension/docs/IntegrationGuide.md [hunk 4]
```diff
@@ -102,7 +102,7 @@ or by asynchronous process like [commercetools-payment-to-order-processor job](h
 ### Validate payment
 There must be at least one CTP payment object of type Adyen
 (`Payment.paymentMethodInfo.paymentInterface = ctp-adyen-integration`)
-AND this CTP payment must have `Payment.interfaceId`.
+AND this CTP payment must have `Payment.custom.fields.merchantReference`.
```
[the remark this answers is not in the document]
Comment (zackpollard) [reply extension/docs/IntegrationGuide.md]: Have now added this, thanks :)

[the remark this answers is not in the document]
Comment (lojzatran) [reply extension/docs/IntegrationGuide.md]: Thanks for the PR :)

**Comment by LEQADA** [comment]
Thank you for the PR @zackpollard . GJ! 🎉

**Pull request merged**